src/sidebar/components/sidebar-panel.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const propTypes = require('prop-types');
+const { createElement } = require('preact');
+const { useEffect, useRef } = require('preact/hooks');
+const scrollIntoView = require('scroll-into-view');
+
+const useStore = require('../store/use-store');
+
+const Slider = require('./slider');
+const SvgIcon = require('./svg-icon');
+
+/**
+ * Base component for a sidebar panel.
+ *
+ * This component provides a basic visual container for sidebar panels, as well
+ * as providing a close button. Only one sidebar panel (as defined by the panel's
+ * `panelName`) is active at one time.
+ */
+function SidebarPanel({ children, panelName, title }) {
+  const panelIsActive = useStore(store => store.isSidebarPanelOpen(panelName));
+  const togglePanelFn = useStore(store => store.toggleSidebarPanel);
+
+  const panelElement = useRef();
+  const panelWasActive = useRef(panelIsActive);
+
+  // Scroll the panel into view if it has just been opened
+  useEffect(() => {
+    if (panelWasActive.current !== panelIsActive) {
```

Comment:
I'm having a little difficulty understanding this logic. It looks like `panelWasActive` is a reference to the old value of the store. Is this a bit like previous props value or something different? 

My understand of `useEffect` is that it will never trigger unless the prop value changes. In this case its set to  `panelIsActive`. So if that is true, then could `panelIsActive` ever not be equal to `panelWasActive.current`?

In order words, If I comment out line 29, and I change the test to 

assert.isTrue(fakeScrollIntoView.calledOnce); it passes. It still gets called once when the component gets created, but it does not get called a second time when we call `wrapper.setProps({});` Or am I thinking about this wrong? I guess I'm really wondering if we need line 29 at all?

Reply:
Well-observed. It may be that there is one too many guarding conditionals here. As you pointed out, assuming `useEffect` really only ever fires if `panelIsActive` has changed, the conditional `if (panelWasActive.current !== panelIsActive)` will always evaluate to `true`.

There are a couple of things that give me pause in ripping out that entire line:

* Communication to future developers. The conditional there expresses the intent: don't fire the scroll unless this is the first render since the panel was inactive. One of the things about (p)react hooks is that they can be a little opaque when reading someone else's code. I feel like this conditional gives a little bit of extra context that could be helpful.
* Trust. While I _think_ `useEffect` will only ever fire if the indicated dependency changes, I'm not sure I'm totally _certain_ of it. Maybe that's me being silly.

Given that the conditional is perhaps useless but is not harmful, I'm a little tempted to leave it in place. Thoughts?